### ISSUE TITLE
Temporary problem found

### ISSUE THREAD

**Issue by q1085909155** [rsedxcftvgyhbujnkiqwe/TF2-Jailbreak-From-Scratch#11]
1、When the player connects to the server, the console reports an error output：

Failed to query (error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'steamid = 'STEAM_0:1:243826165'' at line 1)

2、gang doesn't seem to handle colors correctly, as well as tag
![image](https://github.com/rsedxcftvgyhbujnkiqwe/TF2-Jailbreak-From-Scratch/assets/69096633/10728ed5-9ff0-420e-87bd-3ab69695d959)
![image](https://github.com/rsedxcftvgyhbujnkiqwe/TF2-Jailbreak-From-Scratch/assets/69096633/c9609342-57c1-41aa-a00a-d3efa655f207)
（I have installed the color extension correctly）
3、The Warden menu, which gives the player final request, does not add translation files
4、Suggestion: Will! Help text such as ghelp put it into the menu, otherwise too much text display will cause the player to be dazzled
5、Suggestion: Add a voice control system for the warden, otherwise the red team and the blue team will not make the warden issue instructions
6、Suggestion: Add the jbfs administrator menu to the sourcemod administrator menu, not only through! awm open

ok, those are some of my suggestions and questions, and there may be more I haven't found. Of course, I'll keep testing it for you.

**Comment by q1085909155** [comment]
I created a php page specifically for this purpose, which shows the player who was timeout
![image](https://github.com/rsedxcftvgyhbujnkiqwe/TF2-Jailbreak-From-Scratch/assets/69096633/35ef8dc7-7213-46a7-8814-e58b2b52b608)

Because the plugin setup database section is not set to utf8 format, this caused my game name to be garbled

For timestamps, I recommend that you use the format %Y-%m-%d %H:%M:%S

The specific code is as follows:
`char timestamp[32];
    FormatTime(timestamp, sizeof(timestamp), "%Y-%m-%d %H:%M:%S", GetTime());`
(Of course, I noticed that you said you don't like to provide code directly, these are just my references.)

**Comment by q1085909155** [comment]
I also found a plugin that would make jailbreaking a little more realistic and vivid if possible (prisoners use their fists instead of their melee weapons).
https://forums.alliedmods.net/showthread.php?t=341951

**Comment by q1085909155** [comment]
I don't seem to see anything about Rebel system, and if I did, that would be great

**Comment by rsedxcftvgyhbujnkiqwe** [comment]
Sorry for the late response. 

First and foremost, TF2 just received the 64 bit update - which probably breaks the plugin, as well as sourcemod. I haven't tried it yet, but if it is broken due to 64bit, that will be my priority.

> Failed to query (error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'steamid = 'STEAM_0:1:243826165'' at line 1)

Would you be able to let me know what type of database you're using? I have not run into this issue, and I'm wondering if this is just a problem with the fact that my test server used pgsql. If it is the case, I'll see about making the syntax more universal

> gang doesn't seem to handle colors correctly, as well as tag

That's interesting - another issue that I don't believe I've run into. What sourcemod version and chat processor version are you using?

> The Warden menu, which gives the player final request, does not add translation files

I believe I know why this is occurring. It's set up to use translations but I probably used the wrong translation code and it's running the server language instead of player language.

> Suggestion: Will! Help text such as ghelp put it into the menu, otherwise too much text display will cause the player to be dazzled

I agree, I will see if I can put this into a menu. I'm not entirely happy with it printing to chat, it was a temporary solution to the problem.

> Suggestion: Add a voice control system for the warden, otherwise the red team and the blue team will not make the warden issue instructions

I will see how to add this and try and put it in

> Suggestion: Add the jbfs administrator menu to the sourcemod administrator menu, not only through! awm open

I am unaware how to do this, but I will also try to add it. 

> Because the plugin setup database section is not set to utf8 format, this caused my game name to be garbled

Good catch, I'll fix that 

> (Of course, I noticed that you said you don't like to provide code directly, these are just my references.)

No worries, something like that isn't a problem. Greatly appreciated. I just didn't want entire modules being written, code fixes like that are more than welcome (though at this point, I feel I've learned enough that code contributions would be fine). As for the timestamp, the original idea was to have the unix time in order to allow server owners to handle it on their own, but I agree it's not very readable. Though, there's also the issue of, if you want to sort by date, it could be a problem. Would it be fine to add an extra column with a formatted date? I don't think there's an issue with replacing the column - however for anyone who used the plugin and has to update, it could mess up from an SQL perspective.

> I also found a plugin that would make jailbreaking a little more realistic and vivid if possible

This is an interesting idea. However, it will be low on my priority list for looking into

> I don't seem to see anything about Rebel system, and if I did, that would be great

I considered this a few times, but I never exactly saw the purpose for it. Is the idea behind the implementation that rebel players are 'marked' in order to indicate that they have rebelled, as they are in the other jailbreak plugins? Adding this system is a bit of a pain and will also be low on my priority list, but I will still try to do it.

Once again, thank you for your contributions with the translations. I'll try to put a better credit more centrally on the readme come next update, whenever that is.

**Comment by q1085909155** [comment]
Glad to see your response, after my testing, the 64-bit update doesn't seem to break the plugin's operation.

【Would you be able to let me know what type of database you're using? I have not run into this issue, and I'm wondering if this is just a problem with the fact that my test server used pgsql. If it is the case, I'll see about making the syntax more universal】

I am using mysql5.6

【I considered this a few times, but I never exactly saw the purpose for it. Is the idea behind the implementation that rebel players are 'marked' in order to indicate that they have rebelled, as they are in the other jailbreak plugins? Adding this system is a bit of a pain and will also be low on my priority list, but I will still try to do it.】

Regarding this system, some of the server rules may be related to players with rebel markers, and second, prisoners with rebel markers can be more easily identified by guards, so as not to accidentally injure other friendly prisoners. This is very useful when there are too many players on the server!
In short, the rebel flag is most likely used in conjunction with the server's rules and to facilitate the blue team to identify rebel players.

【Though, there's also the issue of, if you want to sort by date, it could be a problem. Would it be fine to add an extra column with a formatted date?】

emmm, this is just an idea. webphp is just a random action on my part, and few players will notice this page. It was just an idea I had in my spare time and made it happen. About sorting, I didn't really notice how it was sorted. I just combine the page with the database, and the page will display what is in the database table. There should be no sorting

Glad you still maintain the plugin, I will always support it!

**Comment by rsedxcftvgyhbujnkiqwe** [comment]
I am able to start working on this now. First off, I want to try to fix the bugs you've got. I do not experience these so I have to recreate them.

I know what database version you're using, but what sourcemod version? I had been testing this with a development version of sourcemod, (1.12.7054) so knowing which one you're using will help me.

**Comment by q1085909155** [comment]
![image](https://github.com/rsedxcftvgyhbujnkiqwe/TF2-Jailbreak-From-Scratch/assets/69096633/15a5711d-0fd4-42f1-b356-a70df07aeb3c)

Chat color issue I will reinstall my chat processing plugin, maybe the issue will be resolved, you can reduce the priority of this issue

**Comment by rsedxcftvgyhbujnkiqwe** [comment]
I think chat-processor is probably to blame for that issue, because gang chat is the only thing that the chat-processor plugin is even used for.

As for the SQL syntax error, I would try updating to a more recent mysql version perhaps - a quick search shows it's up to version 8. If you still get the error on a different version, I'll set up a server myself and try to recreate it. Though I can tell you right now that syntax wise there shouldn't be any issue with the sql in the plugin since I didn't run into it myself, so the sql server version seems more likely to me.

**Comment by rsedxcftvgyhbujnkiqwe** [comment]
Wow, okay. It's been a while. Partially forgot, partially lost motivation, partially lost interest. 

In the dev branch I have implemented:
-Rebel system
-Warden mute system 
-Help commands moved to actual menus for both the main plugin and the gang plugin
-Plugin commands added to the Sourcemod admin menu
-Datetime column added to gang plugin (see below)
-VScript function changes

As far as I can tell, the issues regarding SQL/Colors were related to your setup. Unless someone can tell me how to actually reproduce those issues, I will assume that's a setup specific bug not something wrong with the plugin.

I went ahead and updated some of the VScript functions to fit my new model for the rebel system. Instead of a "SetRebel" function, there's now "RemoveFreeday" which does as it implies.

Regarding the SQL change for the Gang plugin, I put in some code so that it will check if the new column (or any column) exists or not, and add it if it doesn't. This will help going forward if any more columns get added. No backfilling, though, sorry. 

There's also a fair bit of new translations. If @q1085909155 @RealBerke either of you decide to update translations, I will either put them into this version or a followup one, depending on when I decide to push this update. Testing will likely take me a few days plus I have other life concerns.

**Comment by rsedxcftvgyhbujnkiqwe** [comment]
On second thought - I received the same sql syntax error! I swapped from a postgresql database to mariadb and it happened, so issue officially reproduced. This is a sign that there really is something wrong. I will investigate.

**Comment by rsedxcftvgyhbujnkiqwe** [comment]
I have found and resolved the issue. There was a single space missing, go figure.

**Comment by rsedxcftvgyhbujnkiqwe** [comment]
I scrapped the idea of "fixing" the SQL table. I don't think anyone really uses the plugin much yet, so a manual patch to the sql table can be done instead. I apologize for that but I don't think the epoch time column is even useful so it's getting replaced.

So far everything looking good for the update, running tests now for everything.

**Comment by rsedxcftvgyhbujnkiqwe** [comment]
I believe I have resolved all the problems and requests regarding this issue. Closing it, please make another issue if you have additional suggestions/bufgs.
